airflowHPC/operators/resource_gmx_operator.py

```diff
@@ -70,43 +75,25 @@ def execute(self, context: Context):
                 raise ImportError(
                     "The gmx_executable argument must be set if the gmxapi package is not installed."
                 )
-        bash_path = shutil.which("bash") or "bash"
-        env = self.get_env(context)
-
-        self.log.info(f"mpi_executable: {self.mpi_executable}")
-        self.log.info(f"mpi_ranks: {self.mpi_ranks}")
-        self.log.info(f"gmx_executable: {self.gmx_executable}")
-        self.log.info(f"gmx_arguments: {self.gmx_arguments}")
-        self.log.info(f"input_files: {self.input_files}")
-        self.log.info(f"output_files: {output_files_paths}")
-        self.log.info(f"gpu_ids: {self.gpu_ids}")
-        self.log.info(f"hostname: {self.hostname}")
-
-        self.bash_command = self.create_gmxapi_call(
-            gmx_executable=self.gmx_executable,
-            gmx_arguments=self.gmx_arguments,
-            mpi_executable=self.mpi_executable,
-            mpi_ranks=self.mpi_ranks,
-            input_files=self.input_files,
-            output_files=output_files_paths,
-        )
-        result = self.subprocess_hook.run_command(
-            command=[bash_path, "-c", self.bash_command],
-            stdin=self.stdin,
-            env=env,
-            output_encoding=self.output_encoding,
-            cwd=self.output_dir,
-        )
-
-        if result.exit_code in self.skip_on_exit_code:
-            raise AirflowSkipException(
-                f"Bash command returned exit code {result.exit_code}. Skipping."
-            )
-        elif result.exit_code != 0:
-            raise AirflowException(
-                f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
-            )
-        return result.output
+        rp_endpoint = os.environ.get("RP_ENDPOINT")
+
+        if not rp_endpoint:
+            raise ValueError("RP_ENDPOINT environment variable not set.")
+
+        rp_client = ru.zmq.Client(url=rp_endpoint)
+
+        td = rp.TaskDescription()
+        td.executable     = self.gmx_executable
+        td.ranks          = self.mpi_ranks
+        td.arguments      = self.gmx_arguments
+        td.input_staging  = self.input_files
+        td.output_staging = output_files_paths
+
+        uid = rp_client.request({'cmd': 'rp_execute',
```

Comment:
It took me some time to figure out how to use this request functionality. At any rate doing the following at least causes the command to "run" without error.
```
        uid = rp_client.request(
            cmd="rp_execute",
            task_description=td.as_dict(),
            key=timezone.utcnow().isoformat(),
        )
```
However, all that happens is that a task is created in the rp sandbox with an empty file `task.xxxxxx.exec.sh`.

I could use some insight into how to proceed so that something actually happens. Also note that there is not really a good way to get the key into the task (unless there is some way to pull it from the server running on the executor). At any rate I am not sure if the key is needed.

Is it possible that the rp executor is just not having time to run?

I made a branch where I have been doing testing: https://github.com/ejjordan/airflowHPC/tree/rp-operator-executor

Reply:
Thanks a lot for that feedback, I'll have a look at that branch!